### DISCUSSION TITLE
Setting depth per field

### DISCUSSION THREAD

**Discussion by Ontopic** [payloadcms/payload#207] [Feature Requests & Ideas]
I have a 1-on-1 relationship between two collections that keep an in-sync relationship with each other. A pattern I feel is quite useful. An issues is however that I'm basically getting the full data of the sibling I'm requesting twice, since the other sibling holds a relationship to the requested sibling and is filling that (depending on depth).

What would be a way to limit the depth on certain fields? Or even omit them form results

**Comment by jmikrut** [comment]
Ahhh!!!! Setting max depth on a field basis would be a great feature to have—and best part, it's going to take about 5 minutes for us to implement 💥

Give us a bit here. Might even be today. Thinking the API would look like this:

```js

// Example relationship field config
{
  name: 'someFieldNameHere',
  type: 'relationship',
  relationTo: 'someCollection',
  maxDepth: 1, // new field option
}
```

What do you think?

**Reply by DanRibbens** [reply]
Do you want to call it `maxDepth` or just `depth`, like in the query params?

The idea sounds great.

**Reply by jmikrut** [reply]
Well, `depth` and `maxDepth` are two different things. What I am referring to do is to CAP the max-depth of any given field. So even if someone requests depth=10, that field will only come back populated to its max depth.

**Reply by Ontopic** [reply]
You guys move so quickly! I'm myself getting everything ready for my first Payload staging / production-worthy release, client is very eager. So focussing on that , but all of these ideas sound great and I'm very confident you guys will work this out, but I got to focus now

Gl!

**Reply by DanRibbens** [reply]
https://github.com/payloadcms/payload/pull/208 is ready for review. ~The demo/collections/AllFields~ demo/collections/RelationshipA has a new field with maxDepth on it to try it out.

**Reply by Ontopic** [reply]
Great! Probably the quickest "discussion" ever to be actually solved ;)

Lots of good changes all around and I feel it's just getting started. Maybe I'm going crazy, but even the Payload site seems to have gotten a refreshing shine 😻
